charts/argo-cd/templates/argocd-application-controller/statefulset.yaml

```diff
@@ -13,8 +13,6 @@ metadata:
     {{- include "argo-cd.labels" (dict "context" . "component" .Values.controller.name "name" .Values.controller.name) | nindent 4 }}
 spec:
   replicas: {{ .Values.controller.replicas }}
-  # TODO: Remove for breaking release as history limit cannot be patched
-  revisionHistoryLimit: 5
```

Comment:
If we drop this (and I'd sponsor to do so), we need a major bump and also instructions on howto upgrade.

Reply:
It's mentioned in README that you need to use replace strategy (`helm upgrade --force` or `helm template | kubectl replace -f -`)

Reply:
Helm upgrade force does not work anymore with helmv3+:
- https://github.com/argoproj/argo-helm/issues/970

But ok, we can release that👍

Reply:
@mkilchhofer I'm going to port also #2415 into this as change in Ingress interface will be breaking

Reply:
Changed to configuration in values.yaml to allow upgrade without replacing the resource based on discussion with @mkilchhofer